src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java

```diff
@@ -597,6 +600,41 @@ public void testAddRemoteUrl() throws Exception {
         assertEquals(upstreamRepoURL, gitClient.getRemoteUrl("upstream"));
     }
 
+    @Test (expected = GitException.class)
```

Comment:
I think this is too broad.  Any GitException thrown anywhere in the test method will cause the test to be marked as successful,  even if the exception was in an unexpected location.

The `(expected = GitException.class)` technique is best used for very short tests with very few conditions which could fail or with distinct exceptions rather than overused and generalized exceptions like GitException.
```suggestion
    @Test
```

Reply:
I will make it out. Thanks.